common/lib/util/defaults.js

```diff
@@ -116,16 +127,22 @@ Defaults.normaliseOptions = function(options) {
 	if(!('queueMessages' in options))
 		options.queueMessages = true;
 
-	var production = false;
 	if(options.restHost) {
 		options.realtimeHost = options.realtimeHost || options.restHost;
+		options.fallbackHosts = options.fallbackHostsUseDefault && !options.port && !options.tlsPort ? Defaults.FALLBACK_HOSTS : options.fallbackHosts;
 	} else {
 		var environment = (options.environment && String(options.environment).toLowerCase()) || Defaults.ENVIRONMENT;
-		production = !environment || (environment === 'production');
+		var production = !environment || (environment === 'production');
 		options.restHost = production ? Defaults.REST_HOST : environment + '-' + Defaults.REST_HOST;
 		options.realtimeHost = production ? Defaults.REALTIME_HOST : environment + '-' + Defaults.REALTIME_HOST;
+		if(!options.fallbackHosts && !options.port && !options.tlsPort) {
```

Comment:
Prioroty is given to options.fallbackHosts over options.fallbackHostsUseDefault

Reply:
This is the correct behaviour, no?

Reply:
Actually, I was not sure about the priorities above each other. I mentioned both, so either one of them is right.

Reply:
The problem was logic inconsistency : )

Reply:
@sacOO7 Can you resolve this conversation if this was not an issue, please?

Reply:
As per https://github.com/ably/ably-js/pull/682#discussion_r502371464 there should be no question of priority, either `fallbackHosts` is set and they are the fallback hosts, or `fallbackHostsUseDefault` is set and the default fallbacks are used.

Reply:
Can you mark this as resolved @lmars . I am not able to do it, it seems : (

Reply:
This is addresses in https://github.com/ably/ably-js/pull/689.